src/processes/scoring/calc-karma-score.js

```diff
@@ -14,10 +14,9 @@ const calculateP = (p) => {
 const getUserScore = (user) => {
   return user && typeof user?.user_score === 'number' ? user.user_score : 0;
 };
-const combinedUserScore = async (userId, userScoreCol, sign_user_score) => {
+const combinedUserScore = async (userId, userScoreCol, sign_user_score, anonymousUserId = null) => {
   console.log('Start calculating combined user score');
   const userIds = [userId];
-  const anonymousUserId = await UsersFunction.findAnonymousUserId(userId);
   if (anonymousUserId) {
     userIds.push(anonymousUserId);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `console.log` statement on line 18 is not recommended for production code as it can lead to potential security risks and performance issues. It's better to use a proper logging library that can be configured to log only in development environments or based on log levels.

```diff
- console.log('Start calculating combined user score');
```

Also, the function `combinedUserScore` now accepts an additional parameter `anonymousUserId`. This change might break existing calls to this function elsewhere in the codebase if they are not updated accordingly. Please ensure all references to this function have been updated to pass the new argument.



<!-- This is an auto-generated comment by OSS CodeRabbit -->